plugins/snowflake/dbt/include/snowflake/macros/adapters.sql

```diff
@@ -160,3 +160,33 @@
         {{ column_name }} COMMENT $${{ column_dict[column_name]['description'] | replace('$', '[$]') }}$$ {{ ',' if not loop.last else ';' }}
     {% endfor %}
 {% endmacro %}
+
+
+{% macro get_current_query_tag() -%}
+  {{ return(run_query("show parameters like 'query_tag' in session").rows[0]['value']) }}
+{% endmacro %}
+
+
+{% macro set_query_tag() -%}
+  {% set new_query_tag = config.get('query_tag', none) %}
+  {% if new_query_tag %}
+    {% set original_query_tag = get_current_query_tag() %}
+    {{ log("Setting query_tag to '" ~ new_query_tag ~ "'. Will reset to '" ~ original_query_tag ~ "' after materialization.") }}
+    {% do run_query("alter session set query_tag = '{}'".format(new_query_tag)) %}
+    {{ return(original_query_tag)}}
+  {% endif %}
+  {{ return(none)}}
+{% endmacro %}
+
+{% macro unset_query_tag(original_query_tag) -%}
+  {% set new_query_tag = config.get('query_tag', none) %}
```

Comment:
```suggestion
  {% set new_query_tag = config.get('query_tag') %}
```

There's no reasonable way for you to know this, but `config.get` doesn't do what you think it does! The second argument isn't a default value, but a "validator". See #2441.